### PR TITLE
Marshalling

### PR DESCRIPTION
**Pull request by fbiville**

### CODE REVIEW

File: streaming.md [hunk 1]
```diff
@@ -91,7 +91,8 @@ In particular, the size of the `expectedContentTypes` field (as well as the `out
...
+1. MUST deserialize the *payload* according to the *contentType* field, which MUST be set and represent a concrete MIME type (*i.e.* not a wildcard type). The data structure that corresponds to the result of de-serialization is runtime-specific and is beyond the scope of this specification. To the best extent though, an invoker SHOULD use data structures that are *idiomatic* to the target runtime and pass them to the function if feasible. As an example, if the incoming `contentType` is `application/json` and `payload` contains the bytes of `{"foo": 5}`, then the invoker may use a *map* or a *dictionary* with key `foo`.
```
Comment (glyn) [inline streaming.md]: If the *contentType* field is _not_ set, presumably the invoker is free to do anything. Right? Or should it return some gRPC status error code?

File: streaming.md [hunk 2]
```diff
@@ -143,7 +144,7 @@ message OutputFrame {
 ### OutputFrame
 `OutputFrames` represent data coming out of the function. Upon reception of a result data structure on the j-th output stream of a function, an invoker
 1. MUST serialize the data structure to a series of bytes (hereafter the `OutputFrame payload`) by selecting a MIME type that is compatible with the preference ordered list of MIME types specified in the j-th string of `expectedContentTypes` seen in the [`StartFrame`](#StartFrame). As a consequence, the invoker MUST not send any `OutputFrames` until it has received the `StartFrame`. 
-2. If it cannot successfully serialize data using such a MIME type, then it MUST fail this rpc invocation
+2. If it cannot successfully serialize data using such a MIME type, then it MUST fail this rpc invocation with a crafted status error of code [InvalidArgument](https://github.com/grpc/grpc/blob/2a7191abe717d3ce33b5fcbfd6bd118367499cde/doc/statuscodes.md#status-codes-and-their-use-in-grpc) and the error message MUST start with `"Invoker: Not Acceptable"`. An invoker MAY append error details to that message prefix.
```
Comment (glyn) [inline streaming.md]: How should error details be appended? For instance, should there be punctuation and/or whitespace, e.g. `"Invoker: Not Acceptable: error details"` or is it ok to append the details without any punctuation or whitespace, e.g. `"Invoker: Not Acceptableerror details"`?

File: streaming.md [hunk 3]
```diff
@@ -91,7 +91,8 @@ In particular, the size of the `expectedContentTypes` field (as well as the `out
...
+1. MUST deserialize the *payload* according to the *contentType* field, which MUST be set and represent a concrete MIME type (*i.e.* not a wildcard type). The data structure that corresponds to the result of de-serialization is runtime-specific and is beyond the scope of this specification. To the best extent though, an invoker SHOULD use data structures that are *idiomatic* to the target runtime and pass them to the function if feasible. As an example, if the incoming `contentType` is `application/json` and `payload` contains the bytes of `{"foo": 5}`, then the invoker may use a *map* or a *dictionary* with key `foo`.
+If the *contentType* is not supported, an invoker MUST craft a gRPC status error of code [InvalidArgument](https://github.com/grpc/grpc/blob/2a7191abe717d3ce33b5fcbfd6bd118367499cde/doc/statuscodes.md#status-codes-and-their-use-in-grpc) and the error message MUST start with `"Invoker: Unsupported Media Type"`. An invoker MAY append error details to that message prefix.
```
Comment (glyn) [inline streaming.md]: This is probably off topic for this PR, but what should the invoker do if deserialisation fails?

File: streaming.md (hunk 2 above)
Comment (glyn) [reply streaming.md]: What should the invoker do if serialisation fails?

File: streaming.md (hunk 1 above)
Comment (ericbottard) [reply streaming.md]: To be consistent with the request/reply case, I think we should add that if contentType is not set, it should be assumed to be application/octet-stream

File: streaming.md (hunk 2 above)
Comment (ericbottard) [reply streaming.md]: 1) should add punctuation, but code shouldn't try to chop the prefix IMO
2) the sentence indeed needs to be changed slightly to distinguish between unsupported MIME type(s) and error happening during serialization (two distinct cases). The latter can let the generic gRPC failure bubble up

File: streaming.md (hunk 3 above)
Comment (ericbottard) [reply streaming.md]: idem comment above, this merits being spec'd IMO. The request/reply spec is more explicit than this, so let's align

File: streaming.md [hunk 4]
```diff
@@ -91,10 +91,12 @@ In particular, the size of the `expectedContentTypes` field (as well as the `out
...
+2. If the *contentType* is not supported, an invoker MUST craft a gRPC status error of code [InvalidArgument](https://github.com/grpc/grpc/blob/2a7191abe717d3ce33b5fcbfd6bd118367499cde/doc/statuscodes.md#status-codes-and-their-use-in-grpc) and the error message MUST start with `"Invoker: Unsupported Media Type"`. An invoker MAY append error details to that message prefix. These details SHOULD start with punctuation to separate them from the error message prefix.
+3. MUST fail this rpc invocation if it cannot otherwise successfully deserialize data using such a MIME type
```
Comment (glyn) [inline streaming.md]: Should a particular gRPC status code be used?

File: streaming.md [hunk 5]
```diff
@@ -142,11 +144,12 @@ message OutputFrame {
...
-4. MUST set the `resultIndex` field of the frame to `j`
-5. MAY set additional custom `headers`.
+1. MUST serialize the data structure to a series of bytes (hereafter the `OutputFrame payload`) by selecting a MIME type that is compatible with the preference ordered list of MIME types specified in the j-th string of `expectedContentTypes` seen in the [`StartFrame`](#StartFrame). As a consequence, the invoker MUST not send any `OutputFrames` until it has received the `StartFrame`.
+2. If such a MIME type is not supported, then it MUST fail this rpc invocation with a crafted status error of code [InvalidArgument](https://github.com/grpc/grpc/blob/2a7191abe717d3ce33b5fcbfd6bd118367499cde/doc/statuscodes.md#status-codes-and-their-use-in-grpc) and the error message MUST start with `"Invoker: Not Acceptable"`. An invoker MAY append error details to that message prefix. These details SHOULD start with punctuation to separate them from the error message prefix.
+3. If it cannot otherwise successfully serialize data using such a MIME type, then it MUST fail this rpc invocation
```
Comment (glyn) [inline streaming.md]: Should a particular gRPC status code be used?

Comment (fbiville) [reply streaming.md]: Maybe `UNKNOWN`? WDYT @ericbottard?

File: streaming.md (hunk 4 above)
Comment (fbiville) [reply streaming.md]: Maybe `UNKNOWN`? WDYT @ericbottard?

File: streaming.md (hunk 5 above)
Comment (ericbottard) [reply streaming.md]: Yes, UNKNOWN is the default and seems sensible for this case

File: streaming.md (hunk 4 above)
Comment (fbiville) [reply streaming.md]: done

File: streaming.md (hunk 5 above)
Comment (fbiville) [reply streaming.md]: done

**Review by glyn — APPROVED** [review]
Looks good, thanks!

**Pull request merged**